part-2.py

```diff
@@ -4,13 +4,41 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if len(array) == 0:
+        return False
+    if array[-1] == query:
+        return True
+    return search(array[:-1], query)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    if len(text) == 1 or not text:
+        return True
+    elif text[0] != text[-1]:
+        return False
+    else:
+        return is_palindrome(text[1:-1])
 
 
 # digit_match
-
+import math
+def digit_match(x, y):
```

Comment:
This works, although as we discussed I think it looks cleaner using integer division i.e. `//`